src/Foundatio/Caching/InMemoryCacheClient.cs

```diff
@@ -907,7 +940,7 @@ public void Dispose()
         ItemExpired?.Dispose();
     }
 
-    private class CacheEntry
+    private record CacheEntry
```

Comment:
Made this into a record, I had a hunch this could be causing an issue with ConcurrentDictionary.TryUpdate() as it uses a comparison value with IEquatable... https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.tryupdate?view=net-8.0